src/MySqlConnector/Protocol/Serialization/SocketByteHandler.cs

```diff
@@ -41,30 +39,21 @@ public ValueTask<int> WriteBytesAsync(ArraySegment<byte> data, IOBehavior ioBeha
 			}
 		}
 
-		private async Task<ArraySegment<byte>> DoReadBytesAsync(byte[] buffer, int offset, int count)
+		private async Task<int> DoReadBytesAsync(ArraySegment<byte> buffer)
 		{
-			m_socketAwaitable.EventArgs.SetBuffer(buffer, offset, count);
+			m_socketAwaitable.EventArgs.SetBuffer(buffer.Array, buffer.Offset, buffer.Count);
 			await m_socket.ReceiveAsync(m_socketAwaitable);
-			return new ArraySegment<byte>(buffer, 0, m_socketAwaitable.EventArgs.BytesTransferred);
+			return m_socketAwaitable.EventArgs.BytesTransferred;
 		}
 
 		private async Task<int> DoWriteBytesAsync(ArraySegment<byte> payload)
 		{
-			if (payload.Count <= m_buffer.Length)
-			{
-				Buffer.BlockCopy(payload.Array, payload.Offset, m_buffer, 0, payload.Count);
-				m_socketAwaitable.EventArgs.SetBuffer(m_buffer, 0, payload.Count);
-			}
-			else
-			{
-				m_socketAwaitable.EventArgs.SetBuffer(payload.Array, payload.Offset, payload.Count);
-			}
+			m_socketAwaitable.EventArgs.SetBuffer(payload.Array, payload.Offset, payload.Count);
```

Comment:
The previous code was intentional: d705e209dd34f21cab4eeb4b036427adf37c3d11

I wonder if keeping `m_buffer` in this new code and copying in/out of it would be an improvement or not.